controllers/vspherecluster_reconciler.go

```diff
@@ -117,18 +118,90 @@ func (r *clusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return r.reconcileDelete(ctx, clusterContext)
 	}
 
-	if cluster == nil {
-		log.Info("Waiting for Cluster controller to set OwnerRef on VSphereCluster")
-		return reconcile.Result{}, nil
-	}
-
 	// Handle non-deleted clusters
 	return r.reconcileNormal(ctx, clusterContext)
 }
 
+// patch updates the VSphereCluster and its status on the API server.
+func (r *clusterReconciler) patch(ctx context.Context, clusterCtx *capvcontext.ClusterContext) error {
+	// always update the readyCondition.
+	conditions.SetSummary(clusterCtx.VSphereCluster,
+		conditions.WithConditions(
+			infrav1.VCenterAvailableCondition,
+		),
+	)
+
+	if !v1beta2conditions.Has(clusterCtx.VSphereCluster, infrav1.VSphereClusterVCenterAvailableV1Beta2Condition) {
+		v1beta2conditions.Set(clusterCtx.VSphereCluster, metav1.Condition{
+			Type:    infrav1.VSphereClusterVCenterAvailableV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  infrav1.VSphereClusterVCenterNotAvailableV1Beta2Reason,
+			Message: "did not observe VCenterAvailable status",
+		})
+	}
+
+	if !v1beta2conditions.Has(clusterCtx.VSphereCluster, infrav1.VSphereClusterClusterModulesReadyV1Beta2Condition) {
+		v1beta2conditions.Set(clusterCtx.VSphereCluster, metav1.Condition{
+			Type:    infrav1.VSphereClusterClusterModulesReadyV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  infrav1.VSphereClusterClusterModulesNotReadyV1Beta2Reason,
+			Message: "did not observe cluster module status",
+		})
+	}
+
+	if err := v1beta2conditions.SetSummaryCondition(clusterCtx.VSphereCluster, clusterCtx.VSphereCluster, infrav1.VSphereClusterReadyV1Beta2Condition,
+		v1beta2conditions.ForConditionTypes{
+			infrav1.VSphereClusterFailureDomainsReadyV1Beta2Condition,
```

Comment:
q: should we make this condition optional (IgnoreIfMissing) considering we don't have a fallback set above?